### PR TITLE
Add Rewards claim command

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

### What Changed?
- Add rewards claim command

### TODO
- [x] Add private key signer for simple testing - 88433c312c3b69cae061990bd0fbb55873194cd2
- [x] Add other signers for support a02839271e8fd7f4c770836634af730c90c23a10 (Fireblocks/Web3 in separate PR after this is merged)
- [x] Fix the env / network stuff 0e2e7104d958fa7fd855fd5a3269104aaf1c4281
- [ ] Implement Output flag
- [x] After a successful claim, log a message 3f4c90eabf7687202bf72f394794546b37e222de
- [x] Wrap errors properly 5fd6defce07538385fce67bbc22fe8efe42decc8
- [x] Implement password with echo command for automation - this already works with the cli
- [x] Add better log messages while generating claims 3f4c90eabf7687202bf72f394794546b37e222de
- [x] For non broadcast scenario, output a good message that can be used for someone to send tx themselves fc78a3d9ed00c21f30a6d82d19ca7d6272bff2f9

Output when broadcast is disabled.
Screenshot 2024-07-03 at 10 57 03 PM

### CODE REVIEW

File: pkg/common/flags/general.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package flags
+
+import "github.com/urfave/cli/v2"
+
+var (
+	NetworkFlag = cli.StringFlag{
+		Name:    "network",
+		Aliases: []string{"n"},
+		Usage:   "Network to use. Currently supports 'preprod', 'holesky' and 'mainnet'",
```
Comment (seanmcgary) [inline pkg/common/flags/general.go:9]: Networks should be either `holesky` or `mainnet`. You'll need an `environment` flag that could have: `preprod`, `testnet` or `prod`

File: pkg/rewards/claim.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+	contractrewardscoordinator "github.com/Layr-Labs/eigenlayer-contracts/pkg/bindings/IRewardsCoordinator"
+
+	"github.com/Layr-Labs/eigenlayer-rewards-proofs/pkg/claimgen"
+	"github.com/Layr-Labs/eigenlayer-rewards-proofs/pkg/proofDataFetcher/httpProofDataFetcher"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
+	rewardscoordinator "github.com/Layr-Labs/eigensdk-go/contracts/bindings/IRewardsCoordinator"
```
Comment (seanmcgary) [inline pkg/rewards/claim.go:26]: This and line 18 _should_ be the same, so not sure why you need both. Line 18 is the correct one to use

File: pkg/rewards/claim.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+		RewardsCoordinatorAddress: gethcommon.HexToAddress(rewardsCoordinatorAddress),
+		ChainID:                   chainID,
+		ProofStoreBaseURL:         proofStoreBaseURL,
+		Environment:               env,
+		RecipientAddress:          recipientAddress,
+		SignerConfig:              *signerConfig,
+	}, nil
+}
+
+func getEnvAndNetwork(network string) (string, string) {
```
Comment (seanmcgary) [inline pkg/rewards/claim.go:293]: Im guessing this is taken from the rewards-updater, which is kinda out of date.

This is the correct list of environments: https://github.com/Layr-Labs/proofs-service/blob/ab56d567f5ec759103e2a95488d278431729e75e/internal/config/config.go#L35-L46

File: pkg/utils/constants.go [hunk 4]
```diff
@@ -12,4 +14,14 @@ const (
 	MainnetChainId = 1
 	HoleskyChainId = 17000
 	LocalChainId   = 31337
+
+	MainnetNetworkName = "mainnet"
+	HoleskyNetworkName = "holesky"
+	PreprodNetworkName = "preprod"
```
Comment (seanmcgary) [inline pkg/utils/constants.go:20]: should also be `holesky`. Network would be either `preprod` or `testnet`

File: pkg/common/flags/general.go (hunk 1 above)
Comment (shrimalmadhur) [reply pkg/common/flags/general.go:9]: fair point.

File: pkg/rewards/claim.go (hunk 2 above)
Comment (shrimalmadhur) [reply pkg/rewards/claim.go:26]: basically what's happening here is - eigenlayer-cli uses `eigensdk` bindings (which we use heavily in CLI for reading/writing onchain) and rewards proof is using eigenlayer-contracts bindings. I have talked to @samlaf about using same eventually but till then we need to import to have conversion between contracts bindings and eigensdk bindings

File: pkg/rewards/claim.go (hunk 3 above)
Comment (shrimalmadhur) [reply pkg/rewards/claim.go:293]: yea I am going to clean this up a bit. env and network is confusing

File: pkg/common/flags/general.go (hunk 1 above)
Comment (shrimalmadhur) [reply pkg/common/flags/general.go]: updated 0aeb8f1a8894dac95c5306f67f45187b8d4b0e95

File: pkg/rewards/claim.go (hunk 3 above)
Comment (shrimalmadhur) [reply pkg/rewards/claim.go]: clear now 0aeb8f1a8894dac95c5306f67f45187b8d4b0e95

File: pkg/utils/constants.go (hunk 4 above)
Comment (shrimalmadhur) [reply pkg/utils/constants.go]: updated 0aeb8f1a8894dac95c5306f67f45187b8d4b0e95

**Review by seanmcgary — APPROVED** [review]

**Pull request merged**